src/pipe.rs

```diff
@@ -77,10 +77,43 @@
 use std::io::{Error, ErrorKind};
 use std::os::unix::io::{AsRawFd, IntoRawFd, RawFd};
 
-use libc::{self, c_int};
+use libc::{self, c_int, siginfo_t};
 
 use crate::SigId;
 
+/// nfx: Document it!
+pub trait ExtractAndBuffer: Clone + Copy {
+    /// nfx: Document it!
+    fn new(source: &siginfo_t) -> Self;
+    /// nfx: Document it!
+    fn to_buffer(&self, buffer: &mut [u8]) -> u8 {
+        let self_size = std::mem::size_of::<Self>();
+        if self_size < buffer.len() {
+            let src = self as *const _ as *const u8;
+            let dst = &mut buffer[..] as *mut _ as *mut u8;
+            unsafe { std::ptr::copy_nonoverlapping(src, dst, self_size) };
```

Comment:
`src` and `dst` are `*u8` so alignment for the copy is not an issue.  `buffer` is only ever accessed as a reference or pointer to `u8` so that's good.  `self` is correctly aligned because the compiler created it (on the stack) and it's only accessed as a `*u8` for the copy.  That should take care of all alignment mistakes.